django_embed_template/templatetags/embed.py

```diff
@@ -39,11 +39,9 @@ def render(self, context):
         # The next lines derive from django.template.base.ExtendsNode.render()
         compiled_parent = self.extend_node.get_parent(context)
 
-        if BLOCK_CONTEXT_KEY not in context.render_context:
-            context.render_context[BLOCK_CONTEXT_KEY] = BlockContext()
-        block_context = context.render_context[BLOCK_CONTEXT_KEY]
-
-        old_blocks = block_context.blocks.copy()
+        # Ignore parent blocks
+        block_context = BlockContext()
+        context.render_context[BLOCK_CONTEXT_KEY] = block_context
```

Comment:
The problem with starting afresh and not restoring the old value is that you're modifying the block context in the outer scope as well. If block names are valid outside the `{% embed %}` context (or even in another `{% embed %}`, this causes weird behavior. E.g. if `{% block 'A' %}` is overridden in the first `embed`, but not in a later context, it appears as if it was added in the later context as well.